src/graphql/resolvers.js

```diff
@@ -58,6 +99,78 @@ export default {
         discountCode: errors.length === 0 ? code : null,
         errors
       };
+    },
+
+    createContributor: async (_, { input }) => {
+      const shopifyCustomerID = await createShopifyCustomer(input);
+
+      const [
+        { totalContributions: contributionCount, pullRequests }
+      ] = await Promise.all([
+        getContributorInfo(input.githubUsername),
+        prisma.createContributor({
+          githubUsername: input.githubUsername,
+          email: input.email,
+          shopifyCustomerID
+        })
+      ]).catch(error => {
+        // @todo: more useful error logging
+        throw new ApolloError(error.message);
+      });
+
+      return {
+        githubUsername: input.githubUsername,
+        email: input.email,
+        shopifyCustomerID,
```

Comment:
We should add a comment here explaining that this is only returned for use in child resolvers.